### ISSUE TITLE
Missing Help text when Docking Module unavailable

### ISSUE THREAD

**Issue by rnyoakum** [raceintospace/raceintospace#288]
The game allows players to schedule certain docking missions (Ex: Mission 29, ORBITAL-OEX) which do not include a Docking Module as payload. The player is supposed to use a DM already in orbit. In these cases, when the player approves hardware assembly the VAB provides a warning message `Help(i155);`. The message alerts the player that the mission does not include a DM, but the player may have another flight which will put one in orbit. If the player proceeds with the mission, and does not have a DM in orbit when it is time to launch, the game is supposed to automatically downgrade the mission to one without a docking step (this works) and provide a Help notification (this does not).

In the code - newmis.cpp / MisAnn(char plr, char pad) - there is clearly a variable HelpFlag that is set when the mission is downgraded. Later, there is a check
```
    if (HelpFlag) {
        Help("i156");    // Notification of being an Help
    }
```
intended to display the alert. Yet there is no such help message. I looked at help.cdr using xxd, and it lacks a header entry for "I156". Reading all the help messages, I didn't see any text that fit this situation. I considered if it was a typo of "I155" as that's the VAB docking warning, but the VAB text doesn't work; the "I155" text explicitly states that it is just a warning on an upcoming mission.

If anyone remembers earlier versions of BARIS, was there a text popup in this situation?

In any case, there **should** be a help message explaining that the mission has been downgraded & why. Currently, the game doesn't include the proper message, so this probably needs to wait until the help.cdr file has been converted to a more malleable format.

**Comment by peyre** [comment]
That's a great catch!  I agree, a warning about it would be a good thing.

You know, I also think it would be handy if we could display somewhere whether you currently have a DM in orbit, and if so for how much longer it'll be available.  Can't think where would be good to put that, though.

**Comment by rnyoakum** [comment]
That's a really good idea, and I'm not sure where to put it either. Another concern is that Mission 29 doesn't allow the player to include a DM, even though it's required. If DMs persist in orbit, whether the mission hardware includes a DM should be configurable on all applicable docking missions. One time I was playing and had a series of unmanned docking missions, and as I recall every mission required a DM even though I supposedly could reuse them. But that's probably a trickier feat of programming.

**Comment by peyre** [comment]
The ability to add a DM to an (Orbit) mission is a good idea, but yes I see the difficulty too.  Ideally I would suggest that the mission could be downgraded to a regular docking mission, but downgrades don't have a mechanism for adding hardware (the DM in this case).

I think it makes a lot of sense to enable adding a DM - it's a minimal change of plans for the mission, realistically, and it would make me much less hesitant to schedule an (Orbit) mission.  **If** you can think of a way to do it.

**Comment by rnyoakum** [comment]
I think I was unclear - I meant making a DM configurable on the mission loadout in the VAB.

**Comment by peyre** [comment]
Oh!  That would be an elegant way to do it, from the player's POV.  I guess toggling it on/off would have to change the mission type, which would be an odd exception to write into the code, but it would be a nice feature if we can do it.

**Comment by rnyoakum** [comment]
I mentioned this issue to my brother, who volunteered to write a Help.cdr <-> text conversion program. If there's any Help messages you would like to see added, let me know. There should be text written for

- this issue (#288)
- End of Turn warnings (#166)
- If a planned duration is invalid for a future mission (#278)
- If you try to Continue without recruiting enough astronauts/cosmonauts

If you'd like to write the text for any of these, let me know. If not, I'll try to write something.

**Comment by rnyoakum** [comment]
I pushed a commit (986cc1bcdb7a983107d668c7b642cb0e1b2443b3) that should handle:

* Creates the help message "I156" for when a lack of a DM requires a mission to be scrubbed at launch, and adds a new message "I157" for when it's automatically downgraded instead.
* Resolves issue #278 by creating help message "I160" for when a player tries to create a Duration mission with No Duration or Duration A.
* Creates distinct VIB/VAB versions ("I158", "I159") of the EVA/Docking requirement warnings used in Future Missions ("I118", "I119"), per issue #277.
* Creates a more detailed End-of-Turn warning per https://github.com/raceintospace/raceintospace/issues/166#issuecomment-624120032.

Hopefully, the text is clear enough. I was unsure of what colors to use in the Help entries; the help.cdr file didn't come with guidelines. I forgot to handle

* If you try to Continue without recruiting enough astronauts/cosmonauts

but the help.cdr editor is working quite well. The tool comes as multiple python files, so I didn't include it in the src/utils folder. If the help.cdr file needs more frequent updating, that will change. Until then, if anyone would like their own copy of the editor, let me know :).
